### PR TITLE
Plugins: Combine local and remote plugins into one structure

### PR DESCRIPTION
**Pull request by mckn**
**What this PR does / why we need it**:
Currently the plugin catalog pulls plugin info from two endpoints (gcom and Grafana app backend) however their schemas are different and this creates a few complications with the features we'd like to implement in the frontend.

This PR refactors the catalog to cater for this issue by mapping / munging the data from each endpoint into a schema that caters for the Plugin Catalog UI.

It also fixes an issue with the SearchField where useDebounce triggers on first render causing the filters to reset to `all`.

**Which issue(s) this PR fixes**:

Related #36229

**Special notes for your reviewer**:
Further clean up and more tests will be written once this is merged and work continues on https://github.com/grafana/grafana/pull/36855 and https://github.com/grafana/grafana/pull/36719 which are currently blocked by this PR.

### CODE REVIEW

File: public/app/features/plugins/admin/api.ts [hunk 1]
```diff
@@ -63,22 +62,11 @@ async function uninstallPlugin(id: string) {
   return await getBackendSrv().post(`${API_ROOT}/${id}/uninstall`);
 }
 
-async function updatePlugin(pluginId: string, data: Partial<PluginMeta>) {
```
Comment (jackw) [inline public/app/features/plugins/admin/api.ts:66]: This is redundant and  left over from when the Catalog was a Grafana app plugin and had a settings page.

File: public/app/features/plugins/admin/components/SearchField.tsx [hunk 2]
```diff
@@ -1,19 +1,39 @@
-import React, { useState } from 'react';
+import React, { useState, useRef } from 'react';
 import { css } from '@emotion/css';
 import { useStyles2 } from '@grafana/ui';
 import { GrafanaTheme2 } from '@grafana/data';
 import { useDebounce } from 'react-use';
+// import { useDebounce } from 'react-use';
```
Comment (ashharrison90) [inline public/app/features/plugins/admin/components/SearchField.tsx:6]: can remove this line now 👍

File: public/app/features/plugins/admin/components/InstallControls.tsx [hunk 3]
```diff
@@ -173,26 +171,5 @@ export const getStyles = (theme: GrafanaTheme2) => {
...
-        margin-bottom: ${theme.spacing(2)};
-      }
-
-      li {
-        margin-left: ${theme.spacing(2)};
-        & > p {
-          margin: ${theme.spacing()} 0;
-        }
-      }
-    `,
```
Comment (ashharrison90) [inline public/app/features/plugins/admin/components/InstallControls.tsx:196]: nice spot 🙌

**Review by ashharrison90 — APPROVED** [review]
i don't really know much about the plugin architecture, but the changes all make logical and syntactical sense so that's good 😄 

one tiny commented out line you can remove, otherwise lgtm 👍

**Review by wbrowne — APPROVED** [review]
From a functionality point of view - LGTM!

**Pull request merged**